core/chain_makers.go

```diff
@@ -424,10 +424,10 @@ func GenerateVerkleChain(config *params.ChainConfig, parent *types.Block, engine
 		}
 		return nil, nil
 	}
+	triedb := state.NewDatabaseWithConfig(db, nil)
+	triedb.EndVerkleTransition()
 	var snaps *snapshot.Tree
 	for i := 0; i < n; i++ {
-		triedb := state.NewDatabaseWithConfig(db, nil)
-		triedb.EndVerkleTransition()
```

Comment:
removing `EndVerkleTransition` from the loop is incorrect in case the status of `ended` is linked to the root - because then the status has to be marked for every subsequent block. Clearly not a good thing.

Note to self: also write the end and start block number, so that we can tell if a block is in an ended state. We need to pass the block number all over the place and that is complicated. A better approach, to investigate in a different PR, might consist in storing the ending status in the statedb proper, so that you can access is from e.g. `OpenTrie` without having to pass it the block number.